File: 2st/Riyeon/가장 가까운 같은 글자.java
class Solution {
    public int[] solution(String s) {
        int[] answer = new int[s.length()];
        answer[0] = -1;
        for(int i=1; i<answer.length; i++){
            answer[i] = -1;
            for(int j=i-1; j>=0; j--)
                if(s.charAt(i) == s.charAt(j)){
                    answer[i] = i-j;
                    break;
                }
        }
        return answer;
    }
}

File: 2st/Riyeon/가장 가까운 같은 글자.py
def solution(s):
    answer = []
    temp = []
    for i in s:
        if i not in temp:
            answer.append(-1)
            temp.append(i)
        else:
            temp.append(i)
            for j in range(1,len(temp)):
                if temp[len(temp)-1-j] == i:
                    answer.append(j)
                    break
    return answer